server/stream.go

```diff
@@ -4851,32 +4851,34 @@ func (mset *stream) getDirectRequest(req *JSApiMsgGetRequest, reply string) {
 			return
 		}
 
-		hdr := sm.hdr
 		ts := time.Unix(0, sm.ts).UTC()
-
-		if isBatchRequest {
-			if len(hdr) == 0 {
-				hdr = fmt.Appendf(nil, dgb, name, sm.subj, sm.seq, ts.Format(time.RFC3339Nano), np, lseq)
-			} else {
-				hdr = copyBytes(hdr)
-				hdr = genHeader(hdr, JSStream, name)
-				hdr = genHeader(hdr, JSSubject, sm.subj)
-				hdr = genHeader(hdr, JSSequence, strconv.FormatUint(sm.seq, 10))
-				hdr = genHeader(hdr, JSTimeStamp, ts.Format(time.RFC3339Nano))
-				hdr = genHeader(hdr, JSNumPending, strconv.FormatUint(np, 10))
-				hdr = genHeader(hdr, JSLastSequence, strconv.FormatUint(lseq, 10))
-			}
-			// Decrement num pending. This is optimization and we do not continue to look it up for these operations.
-			np--
-		} else {
-			if len(hdr) == 0 {
-				hdr = fmt.Appendf(nil, dg, name, sm.subj, sm.seq, ts.Format(time.RFC3339Nano))
+		var hdr []byte
+		if !req.NoHeaders {
```

Comment:
Should support for this also be added in `getDirectMulti`?

Reply:
I thought for multi it would be required to know the subject of the retrieved messages - which comes in headers. So being cautious and only adding it to the one message variant for now

Reply:
It could be but seems minimal benefit then, for counters - which we really need this for - a single no headers get is going to bring big benefits I think, but doing some partial headers thing mostly wouldn't help imo